### PR TITLE
Variable interpolation in pipeline steps

### LINKED ISSUE #320: Allow parametrized queries
As a query conceptor, I would like to be able to use templated strings rather than hardcoded values. 

Using [lodash template syntax](https://lodash.com/docs/4.17.15#template), this
could look like:

```json
[
  {
    "name": "domain",
    "domain": "team_infos",
  },
  {
    "name": "filter",
    "condition": {
      "and": [
        "operator": "gt",
        "column": "age",
        "value": "<%= someobj.age %>"
      ]
    }
  }
]
```

### PR DESCRIPTION
**Pull request by adimascio**
Implement parametrized queries.

Closes #320

### CODE REVIEW

File: src/lib/templating.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+/**
+ * The `Interpolator` class provides an "step interpolator" implementation
+ * for each possible step in a pipeline.
+ * A step interpolator is a function that takes a step in parameter and returns
+ * the "interpolated" version of a step, given a specific interpolate function (e.g.
+ * `_.template`).
+ */
+export class PipelineInterpolator implements StepMatcher<S.PipelineStep> {
```
Comment (vdestraitt) [inline src/lib/templating.ts]: Ok with the list of steps and parameters supporting templating for a first version! Thanks

File: tests/unit/top-step-form.spec.ts [hunk 2]
```diff
@@ -74,6 +76,39 @@ describe('Top Step Form', () => {
     });
   });
 
+  it('should accept templatable values', async () => {
```
Comment (vdestraitt) [inline tests/unit/top-step-form.spec.ts]: Should we add this kind of test in every step test files accepting templating ?

Comment (adimascio) [reply tests/unit/top-step-form.spec.ts]: Maybe but I think all steps (except the `Top` one) already accept strings so there's nothing much to test.

File: src/lib/steps.ts [hunk 3]
```diff
@@ -75,7 +76,7 @@ export type FilterSimpleCondition =
 
 type FilterConditionComparison = {
   column: string;
-  value: number;
+  value: number | string;
```
[the remark this answers is not in the document]
Comment (adimascio) [reply src/lib/steps.ts]: My idea was more to authorize strings natively, not specifically templates. e.g. to filter values `> 'a'`. I even think that we might allow dates. What do you think?

[the remark this answers is not in the document]
Comment (vdestraitt) [reply src/lib/steps.ts]: Ok agreed

File: tests/unit/top-step-form.spec.ts (hunk 2 above)
Comment (vdestraitt) [reply tests/unit/top-step-form.spec.ts]: Oook understood

**Pull request merged**